src/fsharp/NameResolution.fs

```diff
@@ -3253,37 +3273,47 @@ let rec ResolvePartialLongIdentInModuleOrNamespaceForRecordFields (ncenv: NameRe
     // get record type constructors
     let tycons = 
         mty.TypeDefinitions
-        |> List.filter (fun tcref -> not (tcref.LogicalName.Contains(",")))
-        |> List.filter (fun tycon -> tycon.IsRecordTycon)
-        |> List.filter (fun tycon -> not (IsTyconUnseen ad g ncenv.amap m (modref.NestedTyconRef tycon)))
+        |> List.filter (fun tcref -> 
+               (not (tcref.LogicalName.Contains(","))) &&
+               tcref.IsRecordTycon &&
+               (not (IsTyconUnseen ad g ncenv.amap m (modref.NestedTyconRef tcref))))
 
     let ilTyconNames = 
         mty.TypesByAccessNames.Values
-        |> List.choose (fun (tycon:Tycon) -> if tycon.IsILTycon then Some tycon.DisplayName else None)
-        |> Set.ofList
+        |> Seq.choose (fun (tycon:Tycon) -> if tycon.IsILTycon then Some tycon.DisplayName else None)
+        |> Set.ofSeq
     
     match plid with 
     | [] -> 
-        // Collect up the accessible sub-modules 
-       (mty.ModulesAndNamespacesByDemangledName 
-          |> NameMap.range 
-          |> List.filter (fun x -> x.DemangledModuleOrNamespaceName |> notFakeContainerModule ilTyconNames)
-          |> List.filter (fun x -> x.DemangledModuleOrNamespaceName |> IsInterestingModuleName)
-          |> List.map modref.NestedTyconRef
-          |> List.filter (IsTyconUnseen ad g ncenv.amap m >> not)
-          |> List.filter (EntityRefContainsSomethingAccessible ncenv m ad)
-          |> List.map ItemForModuleOrNamespaceRef)
-
-       // Collect all accessible record types
-       @ (tycons |> List.map (modref.NestedTyconRef >> ItemOfTyconRef ncenv m) )
-       @ [ // accessible record fields
-            for tycon in tycons do
-                if IsEntityAccessible ncenv.amap m ad (modref.NestedTyconRef tycon) then
-                    let ttype = FreshenTycon ncenv m (modref.NestedTyconRef tycon)
-                    yield! 
-                        ncenv.InfoReader.GetRecordOrClassFieldsOfType(None, ad, m, ttype)
-                        |> List.map Item.RecdField
-         ]
+        let accessibleSubModules =
```

Comment:
For context, imagine the function without any heading comments, forces the reader to scan all the code to get overview.